tetrahedron/src/main/java/com/epam/jwd/additional/TetrahedronFactory.java

```diff
@@ -39,9 +49,12 @@ public List createDotsFromFile() throws IOException {
 
             }
             dotsTetrahedron = dotsForTetrahedron;
-            LOG.info("Tetrahedron's dots: {}",dotsTetrahedron);
-        }
+            LOG.info("Tetrahedron's dots: {}", dotsTetrahedron);
 
-        return dotsTetrahedron;
+            if (!dotsTetrahedron.isEmpty()) {
+                tetrahedrons.add(new Tetrahedron(dotsTetrahedron));
+            }
+        }
+        return tetrahedrons;
```

Comment:
brother, maybe decompose your code - get data from your reader , validate this data and after put in arguments your factory class(method), but i dont think that my opinion 100% true